homeassistant/components/plugwise/climate.py

```diff
@@ -122,49 +141,78 @@ def supported_features(self):
     def device_state_attributes(self):
         """Return the device specific state attributes."""
         attributes = {}
-        attributes["outdoor_temperature"] = self._outdoor_temperature
-        attributes["available_schemas"] = self._api.get_schema_names(
-            self._domain_objects
-        )
-        attributes["active_schema"] = self._active_schema
+        if self._outdoor_temperature:
+            attributes["outdoor_temperature"] = self._outdoor_temperature
+        attributes["available_schemas"] = self._schema_names
+        attributes["selected_schema"] = self._selected_schema
+        if self._boiler_temperature:
+            attributes["boiler_temperature"] = self._boiler_temperature
+        if self._water_pressure:
+            attributes["water_pressure"] = self._water_pressure
         return attributes
 
-    def update(self):
-        """Update the data from the thermostat."""
-        _LOGGER.debug("Update called")
-        self._domain_objects = self._api.get_domain_objects()
-        self._outdoor_temperature = self._api.get_outdoor_temperature(
-            self._domain_objects
-        )
-        self._active_schema = self._api.get_active_schema_name(self._domain_objects)
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list and make the presets with their temperatures available."""
+        return self._presets_list
+
+    @property
+    def hvac_modes(self):
+        """Return the available hvac modes list."""
+        if self._heating_status is not None:
+            if self._cooling_status is not None:
+                return HVAC_MODES_2
+            return HVAC_MODES_1
+        return None
 
     @property
     def hvac_mode(self):
         """Return current active hvac state."""
-        if self._api.get_schema_state(self._domain_objects):
+        if self._schema_status:
             return HVAC_MODE_AUTO
-        return HVAC_MODE_OFF
+        if self._heating_status:
+            if self._cooling_status:
+                return HVAC_MODE_HEAT_COOL
+            return HVAC_MODE_HEAT
+        return None
 
     @property
-    def preset_mode(self):
-        """Return the active preset mode."""
-        return self._api.get_current_preset(self._domain_objects)
+    def thermostat_temperature(self):
```

Comment:
This doesn't seem used. We can remove it.

Reply:
I'm not sure which one can be removed, need to test.

Reply:
The `thermostat_temperature` property is not used. We just use the `self._thermostat_temperature` instance attribute.

Reply:
ok, removed.